gateway/gateway_test.go

```diff
@@ -39,3 +42,198 @@ func TestValidate(t *testing.T) {
 	_, err = conf.Validate("")
 	test.That(t, err, test.ShouldBeError, resource.NewConfigValidationError("", errInvalidSpiBus))
 }
+
+func TestDoCommand(t *testing.T) {
+	g := &Gateway{
+		devices: make(map[string]*node.Node),
+	}
```

Comment:
[q] Is there a reason the gateway is exported? Okay to leave if yes.

Reply:
I can make it private